plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1290,6 +1257,243 @@ public void testJoinPushdown()
+
+        if (!hasBehavior(SUPPORTS_JOIN_PUSHDOWN)) {
+            assertThat(query(session, "SELECT r.name, n.name FROM nation n JOIN region r ON n.regionkey = r.regionkey"))
+                    .isNotFullyPushedDown(joinOverTableScans);
+            return;
+        }
+
+        List<String> nonEqualities = Stream.of(JoinCondition.Operator.values())
+                .filter(operator -> operator != JoinCondition.Operator.EQUAL)
+                //this case tests at the end, because of different plan
+                .filter(operator -> operator != JoinCondition.Operator.IS_DISTINCT_FROM)
+                .map(JoinCondition.Operator::getValue)
+                .collect(toImmutableList());
+
+        // Disable DF here for the sake of negative test cases' expected plan. With DF enabled, some operators return in DF's FilterNode and some do not.
+        Session withoutDynamicFiltering = Session.builder(session)
+                .setSystemProperty("enable_dynamic_filtering", "false")
+                .build();
+
+        // inequality
+        for (String operator : nonEqualities) {
+            // bigint inequality predicate
+            assertThat(query(withoutDynamicFiltering, format("SELECT r.name, n.name FROM nation n JOIN region r ON n.regionkey %s r.regionkey", operator)))
+                    // Currently no pushdown as inequality predicate is removed from Join to maintain Cross Join and Filter as separate nodes
+                    .isNotFullyPushedDown(broadcastJoinOverTableScans);
+
+            assertConditionallyPushedDown(
+                    withoutDynamicFiltering,
+                    format("SELECT r.name, n.name FROM nation n LEFT JOIN region r ON n.regionkey %s r.regionkey", operator),
+                    expectJoinPushdown(operator),
+                    broadcastJoinOverTableScansRepartition);
+
+            assertConditionallyPushedDown(
+                    withoutDynamicFiltering,
+                    format("SELECT r.name, n.name FROM nation n RIGHT JOIN region r ON n.regionkey %s r.regionkey", operator),
+                    expectJoinPushdown(operator),
+                    broadcastJoinOverTableScansRepartition);
+
+            assertConditionallyPushedDown(
+                    withoutDynamicFiltering,
+                    format("SELECT r.name, n.name FROM nation n FULL JOIN region r ON n.regionkey %s r.regionkey", operator),
+                    expectJoinPushdown(operator) && hasBehavior(SUPPORTS_JOIN_PUSHDOWN_WITH_FULL_JOIN),
+                    broadcastJoinOverTableScansRepartition);
+        }
+
+        // IS DISTINCT FROM
+        String distinctOperator = "IS DISTINCT FROM";
+        assertConditionallyPushedDown(
+                withoutDynamicFiltering,
+                format("SELECT r.name, n.name FROM nation n LEFT JOIN region r ON n.regionkey %s r.regionkey", distinctOperator),
+                expectJoinPushdown(distinctOperator),
+                broadcastJoinOverTableScans);
```

Comment:
different than two below (broadcastJoinOverTableScansRepartition)

this 3 assertions are reused many times in the same way. May be extract?

Reply:
not sure, it's kind of test code, which I'd like to read and see immediately what the query is executed, not to jump to another method to see what is there and then come back

Reply:
separate test methods removed